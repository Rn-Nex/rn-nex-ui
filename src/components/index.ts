export * from './Accordion';
export * from './ActivityIndicator';
export * from './Avatar';
export * from './Badge';
export * from './Box';
export * from './Button';
export * from './Card';
export * from './CheckBox';
export * from './Chip';
export * from './Dialog';
export * from './Divider';
export * from './DropDown';
export * from './Image';
export * from './List';
export * from './Pagination';
export * from './Portal';
export * from './Radio';
export * from './Snackbar';
export * from './Stepper';
export * from './Switch';
export * from './TextField';
export * from './Transitions';
export * from './Typography';
