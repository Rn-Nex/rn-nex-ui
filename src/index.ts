export * from './components';
export * from './libraries';
export * from './utils';
