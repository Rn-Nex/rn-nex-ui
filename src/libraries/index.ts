export * from './themes/v2';
export * from './style';
