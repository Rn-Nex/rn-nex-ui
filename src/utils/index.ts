export * from './styleGenerator';
export * from './randomId';
export * from './utils';
export * from './dimensions';
