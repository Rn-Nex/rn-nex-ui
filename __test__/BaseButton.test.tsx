import { fireEvent, render } from '@testing-library/react-native';
import React from 'react';
import { Text } from 'react-native';
import { BaseButton } from '../src';

describe('Base button component', () => {
  const mockTestId = 'base_button_mock_test_id';

  const mockOnLayout = jest.fn();
  const mockOnPress = jest.fn();
  const mockOnLongPress = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render correctly', () => {
    const { toJSON } = render(<BaseButton />);
    expect(toJSON).toMatchSnapshot();
  });

  it('should render child components correctly', () => {
    const { toJSON } = render(
      <BaseButton>
        <Text />
      </BaseButton>,
    );
    expect(toJSON).toMatchSnapshot();
  });

  it('should call the onLayout function when onLayout function prop passed', () => {
    const { getByTestId } = render(<BaseButton onLayout={mockOnLayout} testID={mockTestId} />);
    fireEvent(getByTestId(mockTestId), 'layout', { nativeEvent: {} });
    expect(mockOnLayout).toHaveBeenCalledTimes(1);
  });

  it('should not trigger onPress or onLongPress when disabled', () => {
    const { getByTestId } = render(
      <BaseButton testID={mockTestId} onPress={mockOnPress} onLongPress={mockOnLongPress} disabled />,
    );

    fireEvent.press(getByTestId(mockTestId));
    fireEvent(getByTestId(mockTestId), 'longPress');

    expect(mockOnPress).not.toHaveBeenCalled();
    expect(mockOnLongPress).not.toHaveBeenCalled();
  });

  it('should have initial scale value of 1', () => {
    const { getByTestId } = render(<BaseButton testID={mockTestId} />);
    const animatedView = getByTestId(mockTestId);

    expect(animatedView.props.style.transform[0].scale).toEqual(1);
  });
});
