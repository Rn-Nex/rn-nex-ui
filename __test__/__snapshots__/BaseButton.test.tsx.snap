// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Base button component should render child components correctly 1`] = `[Function]`;

exports[`Base button component should render correctly 1`] = `[Function]`;
